<a href="https://colab.research.google.com/github/abhishekmishra-bareilly/Machine-learning/blob/main/Deal_with_Missing_Attributes.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# How to deal with Missing Attributes

In [1]:
# Librarry import
import pandas as pd

In [2]:
# connect google drive to run the file
from google.colab import drive #connect with google drive
drive.mount('/content/Drive')

Mounted at /content/Drive


In [3]:
# file path reading or csv upload to coolab from drive
file_path='/content/Drive/MyDrive/Colab Notebooks/Machine Learning/ML Project/data.csv'       
housing = pd.read_csv(file_path)

In [6]:
housing.describe()

CRIM          ZN       INDUS        CHAS         NOX          RM  \
count  506.000000  506.000000  506.000000  506.000000  506.000000  501.000000   
mean     3.613524   11.363636   11.136779    0.069170    0.554695    6.284341   
std      8.601545   23.322453    6.860353    0.253994    0.115878    0.705587   
min      0.006320    0.000000    0.460000    0.000000    0.385000    3.561000   
25%      0.082045    0.000000    5.190000    0.000000    0.449000    5.884000   
50%      0.256510    0.000000    9.690000    0.000000    0.538000    6.208000   
75%      3.677083   12.500000   18.100000    0.000000    0.624000    6.625000   
max     88.976200  100.000000   27.740000    1.000000    0.871000    8.780000   

              AGE         DIS         RAD         TAX     PTRATIO           B  \
count  506.000000  506.000000  506.000000  506.000000  506.000000  506.000000   
mean    68.574901    3.795043    9.549407  408.237154   18.455534  356.674032   
std     28.148861    2.105710    8.707259  168.537116    2.164946   91.294864   
min      2.900000    1.129600    1.000000  187.000000   12.600000    0.320000   
25%     45.025000    2.100175    4.000000  279.000000   17.400000  375.377500   
50%     77.500000    3.207450    5.000000  330.000000   19.050000  391.440000   
75%     94.075000    5.188425   24.000000  666.000000   20.200000  396.225000   
max    100.000000   12.126500   24.000000  711.000000   22.000000  396.900000   

            LSTAT        MEDV  
count  506.000000  506.000000  
mean    12.653063   22.532806  
std      7.141062    9.197104  
min      1.730000    5.000000  
25%      6.950000   17.025000  
50%     11.360000   21.200000  
75%     16.955000   25.000000  
max     37.970000   50.000000

here we see 'RM' have null values because it have only 501 values insted of 506.

**How to deal with Missing Attributes**


1. Get rid of the missing data points means delete these null rows.
2. Get rid of the whole attribute means delete that column.
3. Replace null value with some value like.(0, mean or median)

In [4]:
# 1. Get rid of the missing data points means delete these null rows.
a = housing.dropna(subset=["RM"]) #Option 1
a.shape


(501, 14)

>  In this we delete these rows. now total rows are 501.

In [9]:
# 2. Get rid of the whole attribute means delete that column.
housing.drop("RM", axis=1).shape 


(506, 13)

>  in this we delete the column so only have 13 columns

In [10]:
# 3. Replace null value with some value like.(0, mean or median)
median = housing["RM"].median() 

In [12]:
c= housing["RM"].fillna(median) # Option 3
c.shape

(506,)



> we see in 'RM' column has filled with median





